src/paths.md

```diff
@@ -65,6 +65,7 @@ GenericArgsConst ->
     | LiteralExpression
     | `-` LiteralExpression
     | SimplePathSegment
+    | InferredConst
```

Comment:
This ends up making the grammar ambiguous, because `GenericArgsConst` is in `GenericArg`, and that also contains `Type`, which contains `ParenthesizedType` and `InferredType`.

Not sure how we want to handle that.  What are your thoughts, @ehuss?

Reply:
Possibly also a `ParenthesisedTyOrConst` thing on `GenericArgs` to solve the parens issue too

Reply:
Basically you'd split `GenericArgsConst/Type` into `GenericArgsConst/TypeUnambig/Ambig` and then lift the shared syntax out of the `Ambig` versions and onto `GenericArg` which would also be changed to use the `Ambig` forms